lib/url.js

```diff
@@ -1,6 +1,14 @@
 'use strict';
 
-const punycode = require('punycode');
+function importPunycode() {
+  try {
+    return process.binding('icu');
```

Comment:
Note, may want to look at that flag. I swear it still builds icu even though it isn't linked.